dask_expr/_merge.py

```diff
@@ -515,7 +515,10 @@ def assign_index_merge_transfer(
             index = partitioning_index(index, npartitions)
         df[name] = index
         meta[name] = 0
-        return merge_transfer(df, id, input_partition, npartitions, meta, parts_out)
+        disk = [] if not DASK_GT_20231000 else (True,)
```

Comment:
```suggestion
        disk = () if not DASK_GT_20231000 else (True,)
```

Just me being pedantic.   My apologies.